Corona-Warn-App/src/main/java/de/rki/coronawarnapp/service/submission/ScanResult.kt

```diff
@@ -0,0 +1,34 @@
+package de.rki.coronawarnapp.service.submission
+
+import java.util.regex.Pattern
+
+data class ScanResult(val rawResult: String) {
+    val isValid by lazy {
+        containsValidGUID(rawResult)
+    }
+    val guid: String? by lazy {
+        if (isValid) guid
+        else null
+    }
+
+    private fun matchGUIDPattern(scanResult: String): Boolean {
+        val pattern = QR_CODE_REGEX
+        return pattern.toRegex().matches(scanResult)
+    }
+
+    private fun containsValidGUID(rawResult: String): Boolean {
+        if (rawResult.length > SubmissionConstants.MAX_QR_CODE_LENGTH ||
```

Comment:
Make all constants that are part of the whole `ScanResult` stuff, part of this class such that we have a contained solution.